presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadataFactory.java

```diff
@@ -67,7 +69,8 @@ public HiveMetadataFactory(
             JsonCodec<PartitionUpdate> partitionUpdateCodec,
             TypeTranslator typeTranslator,
             NodeVersion nodeVersion,
-            AccessControlMetadataFactory accessControlMetadataFactory)
+            AccessControlMetadataFactory accessControlMetadataFactory,
+            ThriftHiveMetastoreConfig thriftConfig)
```

Comment:
Using the Thrift metastore configuration here is wrong when we are not using the Thrift metastore. Instead, we could add an `isImpersonationEnabled()` method to `HiveMetastore`, allowing the code to work for any metastore type.